src/main.rs

```diff
@@ -949,6 +1037,77 @@ async fn main() {
         );
     }
 
+    tokio::spawn(async {
+        let local_port = match env::var("LAVA_GITLAB_RUNNER_LOCAL_PORT") {
+            Ok(val) => val
+                .parse()
+                .expect("failed to parse LAVA_GITLAB_RUNNER_LOCAL_PORT as a port number"),
+            Err(_) => {
+                warn!("No LAVA_GITLAB_RUNNER_LOCAL_PORT set, will listen on ephemeral IP.");
+                0u16
+            }
+        };
+
+        let listener = std::net::TcpListener::bind(std::net::SocketAddr::new(
```

Comment:
why bothering with a tcp listener rather then just asking Axum::Server to bind itself